### DISCUSSION TITLE
Manifest resolution issue

### DISCUSSION THREAD

**Discussion by castortech** [eclipse-pde/eclipse.pde#1103] [General]
I am having a very hard time upgrading Jetty, and not sure why.

We have a target platform with jetty 9.4.53 (upgrading from 9.4.40) and 10.0.9 (for Eclipse help)

In the current setup, there is no issue with the import package (see jetty94440). Notice also that in the plug in dependencies we only see 9.4.40
![jetty 94440](https://github.com/eclipse-pde/eclipse.pde/assets/85858/53f4535a-6220-4265-b855-c0b89a6506fe)

Now after updating to jetty 9.4.53, errors are reported from the manifest and it won't compile (see page 1 of pdf). It complains that nothing exports 3 different jetty packages, even if it does AFAICT.
[Jetty 9453 issues.pdf](https://github.com/eclipse-pde/eclipse.pde/files/14117011/Jetty.9453.issues.pdf)

Now if I instead use Required bundle for jetty server and util (see page 2 of pdf), now errors go away and it compiles. Notice also that in the plug in dependencies we can see jetty 10 even if it is excluded by the specific range in bundle-version. Also if I go in the java class and F3 on a jetty class name, it brings me to the source for the 9.4.53 version.

Anyone has seen this kind of mess and what can be done. Been working with PDE for almost 20 years and this is still weird to me.

Thanks
Alain

**Comment by laeubi** [comment]
> Now after updating to jetty 9.4.53, errors are reported from the manifest and it won't compile (see page 1 of pdf). It complains that nothing exports 3 different jetty packages, even if it does AFAICT.

How do you know that? Have you checked in the Target platform state? please note that the package version must not necessarily match the bundle version, so that would be the first to check here.

**Reply by castortech** [reply]
I did check the target platform state and all is well there.

As for version matching, my understanding is that while it doesn't have to match and might choose a newer compatible version, if a range is set, like in my example where the upper range is "10)", I would expect to match up to but not including 10.
